arg_scanner/lib/arg_scanner/state_tracker.rb

```diff
@@ -0,0 +1,145 @@
+require "set"
+require "tempfile"
+require "require_all"
+
+module ArgScanner
+  class StateTracker
+    def initialize
+      at_exit do
+        dir = ENV["ARG_SCANNER_STATE_TRACKER_DIR"]
+        dir = "." if dir.nil? || dir == ""
+        tmp_file = Tempfile.new(["classes-", ".json"], dir )
+        path = tmp_file.path
+        tmp_file.close!
+
+        begin
+          require_all Rails.root.join('lib')
+        rescue => e
+        end
+        begin
+          Rails.application.eager_load!
+        rescue => e
+        end
+
+        File.open(path,"w") { |file| print_json(file) }
+      end
+    end
+
+    private
+    def print_json(file)
+      file.puts("{")
+      print_load_path(file)
+      file.puts(",")
+      print_all_modules(file)
+      file.puts("}")
+    end
+
+    def get_constants_of_class(constants, parent, klass)
+      begin
```

Comment:
You don't need to use `begin..end` in the context of method def. You can write `rescue/ensure` right in the method body